components/welcome-mat/index.jsx

```diff
@@ -0,0 +1,243 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Welcome Mat design pattern](https://lightningdesignsystem.com/components/welcome-mat/) in React.
+// Based on SLDS v2.4.0
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Modal from '~/components/modal';
+import ProgressBar from '~/components/progress-bar';
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+import { WELCOME_MAT } from '../../utilities/constants';
+
+const displayName = WELCOME_MAT;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * whether the modal is open
+	 */
+	isOpen: PropTypes.bool,
+	/**
+	 * **Weclome Mat labels for internationalization**
+	 * This object is merged with the default props object on every render.
+	 * * `title`: Title for the Welcome Mat
+	 * * `description`: Label for the radio input
+	 */
+	labels: PropTypes.shape({
+		title: PropTypes.string,
+		description: PropTypes.string,
+	}),
+	/**
+	 *	Variant of the WelcomeMat
+	 */
+	variant: PropTypes.oneOf([
+		'steps',
+		'info-only',
+		'splash',
+		'trailhead-connected',
+	]),
+	/**
+	 * Link to learn more button
+	 */
+	onRenderInfoActions: PropTypes.func,
+	/**
+	 *  Infobadge for trailhead variant
+	 */
+	infoBadge: PropTypes.func,
```

Comment:
Since this prop should accept only one element of a particular type (WelcomeMatInfoBadge), this PropType should be `PropTypes.node`